packages/web/src/app/Layout.tsx

```diff
@@ -1,14 +1,42 @@
-import { Suspense } from 'react';
-import { Outlet } from 'react-router-dom';
+import { GetLanguagesResponseBody, Language } from '@translation/api-types';
+import { Suspense, useMemo } from 'react';
+import { Outlet, useLoaderData, useOutletContext } from 'react-router-dom';
+import apiClient from '../shared/apiClient';
+import useLocalStorageState from '../shared/useLocalStorageState';
 import Header from './Header';
 
+export function layoutLoader() {
+  return apiClient.languages.findAll();
+}
+
+export interface LayoutContext {
+  language?: Language;
+}
+
+export function useLayoutContext() {
+  return useOutletContext<LayoutContext>();
+}
+
 export function Layout() {
+  const languages = useLoaderData() as GetLanguagesResponseBody;
+  const [language, selectLanguage] = useLocalStorageState<string>(
+    'translation-language',
+    languages.data[0]?.id
+  );
+
+  const outletContext = useMemo<LayoutContext>(
+    () => ({
+      language: languages.data.find((l) => l.id === language),
+    }),
+    [language, languages]
+  );
+
   return (
     <Suspense fallback="loading">
       <div className="min-h-screen flex flex-col">
-        <Header />
+        <Header language={language} onLanguageChange={selectLanguage} />
         <div className="flex-grow relative">
-          <Outlet />
+          <Outlet context={outletContext} />
```

Comment:
We can pass context to the outlet component so that our routes can read the selected language